Konflux readme update (#3231)
konflux-ci/konflux-ci: README.md

# Trying Out Konflux

This section demonstrates the process for deploying Konflux locally, onboarding users,
and building and releasing an application. The procedure contains two options for the
user to choose from for onboarding applications to Konflux:

- Using the Konflux UI
- Using Kubernetes manifests

Each of these options has its pros and cons. The Konflux UI procedure provides a more
streamlined user experience once you finish installing and configuring it. However,
it requires using [Quay.io](https://quay.io) for image registry, plus some additional
initial setup steps compared to using a Kubernetes manifest alone.

**Note:** You can configure your server to operate the same way using either the
Konflux UI method or the terminal and Kubernetes manifests procedure. For example,
you can also configure Konflux to support an image registry using the terminal process.

> In both cases, the recommended way to try out Konflux is using [Kind](https://kind.sigs.k8s.io).

Reviewer: We're keeping line length at 89 characters. That's the reason for the end of this line to be pushed to the next line.

Reviewer: I commented the same for some other lines, but there are more.

This guide will walk you through creating a Kind cluster using the `kind-config.yaml`
manifest, located in the root directory of this repository. This manifest tells
[Kind](https://kind.sigs.k8s.io) to forward port `9443` from the host to the
[Kind](https://kind.sigs.k8s.io) cluster. Kind must configure this port forwarding
to access Konflux.

> **Note**: For security reasons, Konflux limits browser access to the local server.

Author: My understanding is that this is the default setting, and you'll receive a security warning if you attempt to enable remote access. Is this not the case?

If you are using a remote machine to set up Konflux, you'll need to use SSH to set
up port forwarding to access Konflux through your local computer's browser. To do this,
open an additional terminal and type the following command:

```bash

> ssh -L 9443:localhost:9443 <remote_host_user_name>@<remote_host_ip_address>

Reviewer: I believe this won't work until Konflux is up an running. For that reason, it was mentioned later in the document.

```

## Machine Minimum Requirements

> Konflux currently only supports the **x86_64 Linux** platform.

Reviewer: I'm not sure if this statement is true for any deployment of Konflux. The way Konflux is deployed in this context was only tested on x86_64 Linux platforms.

Author: You're right. I only heard of Konflux three weeks ago, so I merely adjusted the existing sentence for clarity. I'll add that it also supports arm64.

The deployment requires the following **free** resources:

- **CPU**: 4 cores
- **RAM**: 8 GB

> **Note**: You will need to add more resources if you plan to run multiple pipelines in parallel.

Author: Thanks! I'll adjust all the line lengths before converting this to a pull request.

## Installing Software Dependencies

:gear: Verify that you have the following applications installed on your host machine

- **git** (v2.46 or newer)
- **golang** (v1.24.6 or newer)
- **[Kind](https://kind.sigs.k8s.io)** (v0.26.0 or newer) along with **podman** (v5.3.1 or newer)
or **docker** (v27.0.1 or newer)
- **kubectl** (v1.31.1 or newer)
- **make** (v4.4.1 or newer)
- **openssl** (v3.2.2 or newer)

## Bootstrapping the Cluster

:gear: Before you can bootstrap and configure your cluster, you will need to clone the Konflux
repository and enter the project directory by typing these commands in your terminal:

```bash
git clone https://github.com/konflux-ci/konflux-ci.git
cd konflux-ci
```

> **Note**: We recommend you increase the `inotify` resource limits to avoid issues with having

Reviewer: ```suggestion
**Note**: We recommend you increase the `inotify` resource limits to avoid issues with
having
```

./deploy-test-resources.sh
```

5. :gear: As a reminder, if you've installed Konflux on a remote host, you will need to configure SSH port
forwarding to access your cluster. To do this, open an additional terminal and run the following command
(make sure to add the details for your remote user and IP address):

```bash

> ssh -L 9443:localhost:9443 <remote_host_user_name>@<remote_host_ip_address>

Reviewer: I think the changes here are fine. I don't think we should include the command at the top of the document, as it may cause confusion there and also become a maintenance burden.

```

6. Once you've set up port forwarding, your cluster's UI will be available at https://localhost:9443.
You can log in using the test user's account.

**username**: **user@konflux.dev**
**password**: **password**

You now have Konflux up and running. Next, we'll configure Konflux to respond to pull request webhooks,
build a user application, and push it to a registry.

## Enable Pipelines Triggering via Webhooks

You can configure Konflux to trigger pipelines anytime someone submits a pull request.

> You can also configure Konflux to send the outcomes of these triggers back to your

Reviewer: Here as well, we need to make sure that this does not come out as an optional step.

This section includes two options for onboarding an application to Konflux.

The first option demonstrates using the Konflux UI to onboard an application
and release its builds to [Quay.io](https://quay.io).

The second option demonstrates the use of Kubernetes manifests to onboard and
release builds to a container registry deployed to the cluster. The idea
behind this scenario is to simplify onboarding and demonstrate Konflux with greater ease.

Both options will use an example repository containing a Dockerfile to be
built by Konflux. Before we get started, you'll need to complete these steps:

1. :gear: Fork the [example repository](https://github.com/konflux-ci/testrepo)
   by clicking its **Fork** button and following the instructions on the **Create a new fork** page

2. :gear: Install the GitHub App you created in the previous section on your fork by doing the following:

    1. Go to your [Applications page on GitHub](https://github.com/settings/apps).

    2. Click on your GitHub App (if you haven't done so yet, follow the instructions on the
    [Create a Pipelines-as-Code GitHub App](https://pipelinesascode.com/docs/install/github_apps/#manual-setup)

>     page.

Reviewer: ```suggestion
    page).
```

    3. Click on the **Install App** link on the left-hand side of the page.

    4. Select your GitHub account (this should be the same account you forked
    the [example repository](https://github.com/konflux-ci/testrepo) to in Step 1).

    5. Select the **Only select repositories** radio button.

    6. Select your forked example repository.

    7. Click the **Install** button.

We will use our Konflux deployment to build and release pull requests for this fork.

### Option 1: Onboard Application with the Konflux UI

With this approach, Konflux can create:

1. Manifests in GitHub for the application pipelines it will run.

2. The [Quay.io](https://quay.io) repositories into which it will push container images.

You can configure the first item by creating

> [GitHub Application Secrets](https://github.com/konflux-ci/konflux-ci/blob/main/docs/github-secrets.md)

Reviewer: Is there a benefit of having the complete URL here comparing to having only the path inside the repository? I think that having the full path is reducing the portability of this document (i.e. if the entire repo is moved).

Now, if you click on the **Components** tab in the Konflux UI, you
should see the component you just created.

**Note**: If you have *NOT* completed the [Quay.io](https://quay.io) setup
steps in the previous section, Konflux will be *UNABLE* to send pull requests
to your repository.

In your GitHub repository, you should now see that Konflux created a pull
request with two new pipelines. Konflux triggers one pipeline when pull request
events occur, such as when you create or modify pull requests, and another
when push events occur, such as when you merge a pull request.

You have successfully onboarded your application, and you can continue to the
[next step](#observe-the-behavior).

### Option 2: Onboard Application with Kubernetes Manifests

With this approach, we use `kubectl` to deploy the manifests for creating
the Application and Component resources, and we manually create the pull
request to configure and run the pipelines using Konflux.

Let's step through the process of deploying a manifest via the terminal.

> 1. :gear: Use your text editor of choice to edit your local copy of the
> [example application manifests](https://github.com/konflux-ci/konflux-ci/blob/main/test/resources/demo-users/user/ns2/application-and-component.yaml).

Reviewer: I think that in this case, contrary to the current and suggested text, it will make sense to have the path to the file visible directly, as this is what the user will need in order to locate the file on their local repo.

```suggestion
1. :gear: Use your text editor of choice to edit your local copy of the
example application manifests at
`test/resources/demo-users/user/ns2/application-and-component.yaml`.
```
